### ISSUE TITLE
Overwritten outputs in MuZeroNetwork

### ISSUE THREAD

**Issue by emmeburgin** [JimOhman/model-based-rl#4]
Submitting this as an issue instead of a pull request because I'm not completely sure I understand the code yet.

It looks to me like a few of the variable assignments in the ``forward`` functions of the prediction and dynamics components of the MuZeroNetwork are being overwritten by successive lines. In ``networks.MuZeroPrediction``:

```
value = F.relu(self.fc_value(out))
value = self.fc_value(out)

policy = F.relu(self.fc_policy(out))
policy = self.fc_policy(out)
```

and in ``networks.MuZeroDynamics``, the same thing:

```
reward = F.relu(self.fc1(out.view(batch_size, -1)))
reward = self.fc2(out.view(batch_size, -1))
```

In this case, variables are first set with ``F.relu`` calls and then immediately overwritten.

Can a developer please clarify which of the following three possibilities is the case here?

1. This code is correct and has some effect I'm not able to identify
2. The code is incorrect but has the intended effect (the relu calls are extraneous and should be removed)
3. The code is incorrect and does not have the intended effect (the variables should not be overwritten, this is a bug)

Thank you!

**Comment by JimOhman** [comment]
Ah, thanks for noticing. This is a remnant of an experiment from a while ago.

The code is incorrect, but had the intended effect (in my experiment), but the ReLU's should not be removed. The reason is that there is another fully-connected layer that was bypassed in an experiment, which gave rise to the hanging ReLU's. I've removed the bypass now.

Thanks again

**Comment by emmeburgin** [comment]
Got it, thanks for your attention on this.

I see that ``reward`` is no longer being overwritten in ``MuZeroDynamics``, but can you clarify the purpose of lines 292 and 295 in ``MuZeroPrediction``? It appears that the function is now using the previously bypassed fully-connected layers (suffixed with ``_o``), but the original outputs from the ReLU's are still being overwritten.

```
292        value = F.relu(self.fc_value(out))
293        value = self.fc_value_o(out)
294
295        policy = F.relu(self.fc_policy(out))
296        policy = self.fc_policy_o(out)
```

**Comment by JimOhman** [comment]
That's just me being tired and not fully fixing it.
292 is supposed to enter 293, as is 295 with 296.
I'll change that, thanks.
